File: cSharp/day22.cs

using System;


Console.WriteLine("Input a number: ");
string stringNum1 = Console.ReadLine();

bool validNumber = 
    int.TryParse(stringNum1, out int num1);

if (validNumber) {
    if (num1 <= 100 && num1 >= 0) {
        Console.WriteLine("Valid number accepted");
    }
    else {
        Console.WriteLine("Not within the range!");
    }
}
else {
    Console.WriteLine("Not a number!");
}

Console.ReadKey();

File: go/day22.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"errors"
)

func main() (bool) {
	fmt.Print("Input a number: ")

	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)

	num, err := strconv.Atoi(input)

	if err == nil {
		if num >= 0 && num<=100 {
			fmt.Println("Valid number accepted");
			return true;
		} else {
			fmt.Println("number not within the range");
			return false;
		}
	} else {
		fmt.Println("Not a number!");
		return false;
	}
	return false;
}